src/main/scala/de/htwg/se/sudoku/model/fileIoComponent/fileIoJsonImpl/FileIO.scala

```diff
@@ -8,53 +8,66 @@ import de.htwg.se.sudoku.model.fileIoComponent.FileIOInterface
-    val json: JsValue = Json.parse(source)
-    val size = (json \ "grid" \ "size").get.toString.toInt
-    val injector = Guice.createInjector(new SudokuModule)
-    size match {
-      case 1 => gridOption = Some(injector.instance[GridInterface](Names.named("tiny")))
-      case 4 => gridOption = Some(injector.instance[GridInterface](Names.named("small")))
-      case 9 => gridOption = Some(injector.instance[GridInterface](Names.named("normal")))
-      case _ =>
-    }
-    gridOption match {
-      case Some(grid) => {
-        var _grid = grid
-        for (index <- 0 until size * size) {
-          val row = (json \\ "row") (index).as[Int]
-          val col = (json \\ "col") (index).as[Int]
-          val cell = (json \\ "cell") (index)
-          val value = (cell \ "value").as[Int]
-          _grid = _grid.set(row, col, value)
-          val given = (cell \ "given").as[Boolean]
-          val showCandidates = (cell \ "showCandidates").as[Boolean]
-          if (given) _grid = _grid.setGiven(row, col, value)
-          if (showCandidates) _grid = _grid.setShowCandidates(row, col)
+
+    Try {
+      val source: String = Source.fromFile(FILE_NAME).getLines.mkString
+
+      val json: JsValue = Json.parse(source)
+      val size = (json \ "grid" \ "size").get.toString.toInt
+      val injector = Guice.createInjector(new SudokuModule)
+      size match {
+        case 1 =>
+          gridOption =
+            Some(injector.instance[GridInterface](Names.named("tiny")))
+        case 4 =>
+          gridOption =
+            Some(injector.instance[GridInterface](Names.named("small")))
+        case 9 =>
+          gridOption =
+            Some(injector.instance[GridInterface](Names.named("normal")))
+        case _ =>
+      }
+      gridOption match {
+        case Some(grid) => {
+          var _grid = grid
+          for (index <- 0 until size * size) {
+            val row = (json \\ "row")(index).as[Int]
+            val col = (json \\ "col")(index).as[Int]
+            val cell = (json \\ "cell")(index)
+            val value = (cell \ "value").as[Int]
+            _grid = _grid.set(row, col, value)
+            val given = (cell \ "given").as[Boolean]
+            val showCandidates = (cell \ "showCandidates").as[Boolean]
+            if (given) _grid = _grid.setGiven(row, col, value)
+            if (showCandidates) _grid = _grid.setShowCandidates(row, col)
+          }
+          gridOption = Some(_grid)
         }
-        gridOption=Some(_grid)
+        case None =>
       }
-      case None =>
+      gridOption
     }
-    gridOption
   }
 
-  override def save(grid: GridInterface): Unit = {
+  override def save(grid: GridInterface): Try[Unit] = {
     import java.io._
-    val pw = new PrintWriter(new File("grid.json"))
-    pw.write(Json.prettyPrint(gridToJson(grid)))
-    pw.close
+
+    Try {
+      val pw = new PrintWriter(new File(FILE_NAME))
+      pw.write(Json.prettyPrint(gridToJson(grid)))
+      pw.close
```

Comment:
not sure if the scala try works the same way but maybe the `PrintWriter` should be closed in a 'finally block'. maybe only Try'ing the write itself? `Try(pw.write(...))`

Reply:
Just creating a new `PrintWriter`-instance can cause e.g. an `FileNotFoundException` which should be also handled so it is not enough to just "trying" the write but indeed we should close the `PrintWriter` if something goes wrong...hmm will figure out how we can archive that...